### PR TITLE
[runtime] Thread MonoError through mono_class_get_overrides_full.

### PR DESCRIPTION
**Pull request by kumpera**

### CODE REVIEW

File: mono/metadata/class.c [hunk 1]
```diff
@@ -4176,11 +4177,8 @@ mono_class_setup_vtable_general (MonoClass *klass, MonoMethod **overrides, int o
 		for (i = 0; i < gklass->vtable_size; ++i)
 			if (gklass->vtable [i]) {
 				MonoMethod *inflated = mono_class_inflate_generic_method_full_checked (gklass->vtable [i], klass, mono_class_get_context (klass), &error);
-				if (!mono_error_ok (&error)) {
-					mono_class_set_type_load_failure (klass, "Could not inflate method due to %s", mono_error_get_message (&error));
-					mono_error_cleanup (&error);
-					return;
-				}
+				if (!is_ok (&error))
```
Comment (luhenry) [inline mono/metadata/class.c]: You have a `goto_if_nok (&error, fail)` helper for that

File: mono/metadata/class.c [hunk 2]
```diff
@@ -4259,20 +4257,20 @@ mono_class_setup_vtable_general (MonoClass *klass, MonoMethod **overrides, int o
...
 		int iface_onum;
-		gboolean ok = mono_class_get_overrides_full (ic->image, ic->type_token, &iface_overrides, &iface_onum, mono_class_get_context (ic));
-		if (ok) {
-			for (int i = 0; i < iface_onum; i++) {
-				MonoMethod *decl = iface_overrides [i*2];
-				MonoMethod *override = iface_overrides [i*2 + 1];
-				if (!apply_override (klass, vtable, decl, override))
-					goto fail;
+		mono_class_get_overrides_full (ic->image, ic->type_token, &iface_overrides, &iface_onum, mono_class_get_context (ic), &error);
+		if (!is_ok (&error))
```
Comment (luhenry) [inline mono/metadata/class.c]: same as https://github.com/mono/mono/pull/6069/files#r152592644

**Comment by kumpera** [comment]
I'll change to use the macro.

**Comment by kumpera** [comment]
Pushed a change to use goto_if_nok

**Comment by lambdageek** [comment]
ARMv5 sf failure is `condition `clause < ji->num_clauses' not met` which was fixed in 13d221bf4010e927ae72c8c955b09929e87dbca5

**Pull request merged**